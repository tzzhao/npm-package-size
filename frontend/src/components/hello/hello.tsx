import {ReactNode} from 'react';
import * as React from "react";

class HelloProps {
  public name?: string = '';
}

export class Hello extends React.Component<HelloProps> {
  render(): ReactNode {
    return <h1>Hello {this.props.name}</h1>;
  }
}
