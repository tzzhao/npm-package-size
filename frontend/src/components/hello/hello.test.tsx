import {render, RenderResult} from '@testing-library/react';
import renderer from 'react-test-renderer';
import * as React from 'react';
import {Hello} from './hello';

test('Sanity test', () => {
  const renderResult: RenderResult = render(<Hello />);
  expect(renderResult.baseElement.textContent).toBe("Hello ");
});

test('Sanity test - SNAPSHOT', () => {
  const component = renderer.create(<Hello />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Test name property', () => {
  const renderResult: RenderResult = render(<Hello name="World" />);
  expect(renderResult.baseElement.textContent).toBe("Hello World");
});

test('Test name property - SNAPSHOT', () => {
  const component = renderer.create(<Hello name="World" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Test html name tag is not interpreted', () => {
  const renderResult: RenderResult = render(<Hello name="World <div>test</div>" />);
  expect(renderResult.baseElement.textContent).toBe("Hello World <div>test</div>");
});

test('Test html name tag is not interpreted - SNAPSHOT', () => {
  const component = renderer.create(<Hello name="World <div>test</div>" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
