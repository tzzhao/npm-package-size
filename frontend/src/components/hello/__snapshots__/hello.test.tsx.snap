// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sanity test - SNAPSHOT 1`] = `
<h1>
  Hello 
</h1>
`;

exports[`Test html name tag is not interpreted - SNAPSHOT 1`] = `
<h1>
  Hello 
  World &lt;div&gt;test&lt;/div&gt;
</h1>
`;

exports[`Test name property - SNAPSHOT 1`] = `
<h1>
  Hello 
  World
</h1>
`;
